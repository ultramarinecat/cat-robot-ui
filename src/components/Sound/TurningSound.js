//
// Turn request sound
//

import React, { memo, forwardRef } from 'react';

const TURNING_SOUND = `${process.env.PUBLIC_URL}/turningSound.mp3`;

export const TurningSound = forwardRef((props, ref) => {
  return (
    <audio ref={ref} data-testid="turning-audio">
      <source src={TURNING_SOUND} preload="auto" />
    </audio>
  );
});

export default memo(TurningSound);
