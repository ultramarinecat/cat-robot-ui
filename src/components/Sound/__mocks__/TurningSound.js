import React, { forwardRef } from 'react';

export const playTurning = jest.fn();
const pause = jest.fn();

const TurningSound = forwardRef((props, ref) => {
  ref.current = {
    play: playTurning,
    pause
  };

  return <div data-testid="turning-audio" />;
});

export default TurningSound;
