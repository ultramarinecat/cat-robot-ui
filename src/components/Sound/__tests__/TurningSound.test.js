import React from 'react';
import { render } from 'react-testing-library';

import { TurningSound, TURNING_SOUND } from '../TurningSound.js';

const TURNING_AUDIO = 'turning-audio';

describe('Turning sound', () => {
  it('should play a turning sound', () => {
    const { container, getByTestId } = render(<TurningSound />);
    const turningAudio = getByTestId(TURNING_AUDIO);

    expect(turningAudio).toBeInTheDocument();
    expect(turningAudio.firstChild).toHaveAttribute('src', TURNING_SOUND);

    expect(container.firstChild).toMatchSnapshot();
  });
});
