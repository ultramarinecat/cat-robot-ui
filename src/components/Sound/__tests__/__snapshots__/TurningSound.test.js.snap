// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Turning sound should play a turning sound 1`] = `
<audio
  data-testid="turning-audio"
>
  <source
    preload="auto"
    src="/turningSound.mp3"
  />
</audio>
`;
